pkg/controller/nodeipam/ipam/multi_subnetworks_handler_test.go

```diff
@@ -0,0 +1,272 @@
+			gotSubnetName, gotSubnetPrefix, err := getSubnetWithPrefixFromURL(tc.url)
+			if (err != nil) != tc.wantErr {
+				t.Errorf("getSubnetWithPrefixFromURL() error = %v, wantErr %v", err, tc.wantErr)
+				return
+			}
+			if gotSubnetName != tc.wantSubnetName {
+				t.Errorf("getSubnetWithPrefixFromURL() gotSubnetName = %v, want %v", gotSubnetName, tc.wantSubnetName)
+			}
+			if gotSubnetPrefix != tc.wantSubnetPrefix {
+				t.Errorf("getSubnetWithPrefixFromURL() gotSubnetPrefix = %v, want %v", gotSubnetPrefix, tc.wantSubnetPrefix)
+			}
+
+		})
+	}
+}
+
+func TestUpdateNodeTopology(t *testing.T) {
+
+	testClusterValues := gce.DefaultTestClusterValues()
+	testClusterValues.SubnetworkURL = "https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1/subnetworks/subnet-def"
+	fakeGCE := gce.NewFakeGCECloud(testClusterValues)
+
+	emptyNodeTopologyCR := &ntv1.NodeTopology{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+		// Status is initially left empty
+	}
+	ntClient := ntfakeclient.NewSimpleClientset(emptyNodeTopologyCR)
+
+	tests := []struct {
+		name            string
+		node            *v1.Node
+		existingSubnets []string
+		wantSubnets     []string
+	}{
+		{
+			name: "node's subnet already exists in the cr",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						testNodePoolSubnetLabelPrefix: "subnet-def",
+						"another-label":               "value",
+					},
+				},
+			},
+			existingSubnets: []string{"subnet-def"},
+			wantSubnets:     []string{"subnet-def"},
+		},
+		{
+			name: "node has a new subnet",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						testNodePoolSubnetLabelPrefix: "new-subnet",
+					},
+				},
+			},
+			existingSubnets: []string{"subnet-def"},
+			wantSubnets:     []string{"subnet-def", "new-subnet"},
+		},
+		{
+			name: "node has a subnet and cr's subnets is empty",
+			node: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						testNodePoolSubnetLabelPrefix: "subnet-def",
+					},
+				},
+			},
+			existingSubnets: []string{},
+			wantSubnets:     []string{"subnet-def"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// populate the def subnet URL
+			// pass the fake nodetopology client
+			// create a fake default cr (outside this loop. at the begining of the test)
+			// populate the existingSubnets in the cr
+			// call the ca.updateNodeTopology and read the subnets
+			// verify the subnets with wantSubnets
+			addSubnetsToCR(tc.existingSubnets, ntClient)
+			ca := &cloudCIDRAllocator{
+				cloud:              fakeGCE,
+				nodeTopologyClient: ntClient,
+			}
+			err := ca.updateNodeTopology(tc.node)
+			if err != nil {
+				t.Errorf("ca.updateNodeTopology() returned error: %v", err)
+			}
+			if !verifySubnetsInCR(t, tc.wantSubnets, ntClient) {
+				t.Errorf("ca.updateNodeTopology() returned incorrect subnets")
+			}
```

Comment:
Add test verification for zone info?

Reply:
We do not plan to use this field for now. We may use it in the future.